Serenity-Website/src/MoodBoard/MoodBoard.jsx

```diff
@@ -1,12 +1,127 @@
+
+};
 
 function MoodBoard (){
+    // set default mood to null
+    const [selectedMood, setSelectedMood] = useState(null);
+    
+    //set current date
+    const [currentDate, setCurrentDate] = useState('');
+    
+    //state hook to set success message for when user submits a mood.
+    const [successMessage, setSuccessMessage] = useState('');
+    
+    //track if mood has been submitted
+    const [moodSubmitted, setMoodSubmitted] = useState(false);
+
+    // set error message
+    const[errorMessage, setErrorMessage] = useState('');
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        // Set the current date and render when the component mounts 
+        const now = new Date();
+        const formattedDate = now.toLocaleString('en-US', 
+            {weekday: 'long', day: 'numeric', month: 'long', hour: 'numeric', minute: 'numeric'});
+        setCurrentDate(formattedDate);
+    }, []);
+
+    const backendUrlAddress = import.meta.env.VITE_BACKEND_ADDRESS
+    const url = `${backendUrlAddress}/moodlog`
+
+    const handleMoodSelect = (mood) => {
+        // handle mood changes
+        setSelectedMood(prevMood => prevMood === mood ? null : mood);
+    }
+
+    const handleMoodSubmit = async() => {
+        if(!selectedMood) {
+            alert('Please select a mood');
+            return;
+        }
+
+        try {
+            const response = await fetch (url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                credentials : 'include',
+                body: JSON.stringify ({ mood: selectedMood })
+            });
+
+            if (response.status === 400){
+                setErrorMessage("Mood already logged for today");
+            } else if (!response.ok) {
+                setErrorMessage("Error logging mood");
+                throw new Error('Failed to log mood');
+            } else {
+                const data = await response.json();
+                setMoodSubmitted(true);
+                // set success message if user logs mood successfully
+                setSuccessMessage('You have successfully logged your mood');
+                // clear error message
+                setErrorMessage('');
+                console.log('Mood loged successfully:', data);
+            }
+            
+            } catch(error) {
+            console.error('Error logging mood', error);
+            setErrorMessage("Error logging mood");
+        }
+    };
+
     return (
         <>
-        <p>Hi there</p>
+        <Header />
+        <div className='back-button-div'><button className='back-button' onClick={() => navigate('/activites-feed')}> <FontAwesomeIcon icon={faX} /> </button></div>
+        <div className='mood-entry'>
+            <h2 className="prompt">How are you feeling today?</h2>
+            <h4 className="current-date"><FontAwesomeIcon icon={faCalendar} />   {currentDate}</h4>
```

Comment:
To make things easier for you (not having to update Date realtime), I think we can take out the date display from the Entry Logging page. It's not crucial to the user experience.